### PR TITLE
Fix: proper extrinsic format and change sr25519 to ed25519

### PR DESCRIPTION
**Pull request by TorstenStueber**
This fixes problems with the Spacewalk integration in Pendulum.

### CODE REVIEW

File: clients/runtime/src/cli.rs [hunk 1]
```diff
@@ -57,8 +58,18 @@ fn get_credentials_from_file(file_path: &str, keyname: &str) -> Result<Pair, Key
     Ok(pair)
 }
 
-pub fn parse_account_keyring(src: &str) -> Result<AccountKeyring, Error> {
-    AccountKeyring::from_str(src).map_err(|_| Error::KeyringAccountParsingError)
+pub fn parse_account_keyring(src: &str) -> Result<Ed25519Keyring, Error> {
```
Comment (TorstenStueber) [inline clients/runtime/src/cli.rs:61]: Unfortunately, `Ed5519Keyring` does not have a `FromStr` implementation (but `Sr25519Keyring` does – strange). I just copied a version here.

File: clients/runtime/src/lib.rs [hunk 2]
```diff
@@ -79,7 +77,7 @@ impl Config for SpacewalkRuntime {
     type Hash = H256;
     type Hashing = BlakeTwo256;
     type AccountId = AccountId;
-    type Address = Self::AccountId;
+    type Address = Address;
```
Comment (TorstenStueber) [inline clients/runtime/src/lib.rs:80]: This is the elementary fix that ensures that subxt uses the actual address format used in the (testchain) runtime and not the account id format.

File: clients/runtime/src/rpc.rs [hunk 3]
```diff
@@ -1,425 +1,410 @@
...
-		const DEFAULT_SPEC_VERSION: RangeInclusive<u32> = 1..=1;
-		const DEFAULT_SPEC_NAME: &str = "standalone";
-	} else if #[cfg(feature = "parachain-metadata")] {
-		const DEFAULT_SPEC_VERSION: RangeInclusive<u32> = 3..=3;
-		const DEFAULT_SPEC_NAME: &str = "parachain";
-		pub const SS58_PREFIX: u16 = 2032;
-	}
+    if #[cfg(feature = "standalone-metadata")] {
+        const DEFAULT_SPEC_VERSION: RangeInclusive<u32> = 1..=1;
+        const DEFAULT_SPEC_NAME: &str = "pendulum-parachain";
```
Comment (TorstenStueber) [inline clients/runtime/src/rpc.rs:22]: Maybe we need to revert this change in order to make the vault usable with the testchain. In this case we should document how to overwrite the spec name on the command line.

File: clients/runtime/src/rpc.rs [hunk 4]
```diff
@@ -1,425 +1,410 @@
...
+    ) -> Result<(), Error> {
+        self.with_unique_signer(|signer| async move {
+            self.api
+                .tx()
+                .spacewalk() // assume that spacewalk pallet is registered in connected chain
+                .redeem(
+                    asset_code.to_vec(),
+                    asset_issuer.to_vec(),
+                    amount,
+                    stellar_vault_pubkey.clone().to_vec(),
```
Comment (TorstenStueber) [inline clients/runtime/src/rpc.rs:402]: We realized in our bug fixing session yesterday that this was missing.

File: clients/runtime/src/rpc.rs [hunk 5]
```diff
@@ -1,425 +1,410 @@
...
+        const DEFAULT_SPEC_VERSION: RangeInclusive<u32> = 3..=3;
+        const DEFAULT_SPEC_NAME: &str = "parachain";
+        pub const SS58_PREFIX: u16 = 2032;
+    }
 }
 const TRANSACTION_TIMEOUT: Duration = Duration::from_secs(300); // 5 minutes
 
-type RuntimeApi =
-	metadata::RuntimeApi<SpacewalkRuntime, SubstrateExtrinsicParams<SpacewalkRuntime>>;
+type RuntimeApi = metadata::RuntimeApi<SpacewalkRuntime, PolkadotExtrinsicParams<SpacewalkRuntime>>;
```
Comment (TorstenStueber) [inline clients/runtime/src/rpc.rs:31]: Changing this from `SubstrateExtrinsicParam` to `PolkadotExtrinsicParam` will ensure that subxt uses an extrinsic format where tips are only value and not pairs of values and asset code.

Quite surprising considering these names.

Comment (gonzamontiel) [reply clients/runtime/src/rpc.rs:31]: wow, curious and weird, nice catch

**Review by gonzamontiel — COMMENTED** [review]
I left one small suggestion, but fine for me otherwise!

**Review by ashneverdawn — APPROVED** [review]
thanks for your comments, really helpful. 
lgtm 👍
